### ISSUE TITLE
Why texture changes frequently in GAN training?

### ISSUE THREAD

**Issue by BaiYu0120** [xinntao/BasicSR#38]
Hi. Have you ever noticed that the detail or texture always changes frequently during training GAN? If yes, why???

**Comment by xinntao** [comment]
Hi @BaiYu0120 
In the beginning, the textures change frequently. 
As the training goes, the results converge and the textures will be more stable among iterations (may also because of a smaller learning rate).

**Comment by BaiYu0120** [comment]
I also found that when the learning rate drops to a very small level, the textures will be stable. But what I worried about is the time when to decay learning rate. Due to texture changing frequently at beginning, sometimes it decay after a good texture, the texture will be stable on a good level. Sometimes it decay after a bad texture, the texture will be stable on a bad level... 
How can I control it, do you have any suggestion?

**Comment by xinntao** [comment]
Hi, @BaiYu0120 
From my experiments, the network always produced good-looking textures (though there are artifacts sometimes, they are more likely from GAN training). Because when the learning rate decreases, the optimization updates will further "refine" the textures, even it falls into a "bad" point. There are not only two states - learning and maintaining the previous results. The model is learning all the time as the learning rate decreases.

You can imagine a general optimization example, where there are local optima and global optima. At the beginning when the learning rate is large, it jumps from here to there and the losses also shake a lot. You may worry what if the model loss is accidentally large at the moment when the learning rate decreases. I do not think the model will then maintain such a large loss with further training of a small learnig rate.

**Comment by BaiYu0120** [comment]
Thank you, @xinntao. I undersand.

**Comment by 924175302** [comment]
@xinntao
Hello, I have a question to ask you, my dataset is not very dependent on textures. I want to reduce the effect of controlling texture in ESRGAN. How can I achieve this?
